libbeat/common/mapval/is_defs.go

```diff
@@ -205,6 +206,26 @@ var IsNonEmptyString = Is("is a non-empty string", func(path path, v interface{}
 	return ValidResult(path)
 })
 
+// IsStringMatching checks whether a value matches the given regexp.
+func IsStringMatching(regexp *regexp.Regexp) IsDef {
+	return Is("is string matching regexp", func(path path, v interface{}) *Results {
```

Comment:
How about printing the regex in the message as well. Like `is string matching '%v'`.

Have you considered to pass the regex as string to `IsStringMatching` and compile it internally?

For matching we also have the package `github.com/elastic/beats/libbeat/common/match`. This one runs some optimizations on the regex speeding up a few use-cases. As this function is mostly used for testing I do not favor one over the other tough.

We could also change the signature to `func IsStringMatching(matcher interface{ MatchString(string) bool }) IsDef` (although this is valid syntax, better give the type a name).

Reply:
the string here is actually not the main one you read in the output, the one on line 221 just a bit below here is the main one a user would see. That one does include the regexp source. We don't interpolate those descriptor strings for the other `Is` definitions in this file, so it'd be weird to do it here.

I'm not super opinionated on it, but it from a UX perspective it's a bit redundant.

Cool stuff about `common/match`. I agree that speed her isn't important and keeping the code simple is probably better.

Reply:
As a side note, I've considered that maybe we should just not even have this initial string description since we have a better one that's emitted with the actual error. I think that's out of scope for this PR however.

Reply:
> As a side note, I've considered that maybe we should just not even have this initial string description since we have a better one that's emitted with the actual error. I think that's out of scope for this PR however.

SGTM.